chia/wallet/puzzles/singleton_top_layer.py

```diff
@@ -0,0 +1,126 @@
+from typing import List, Tuple
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.condition_opcodes import ConditionOpcode
+from chia.types.coin_solution import CoinSolution
+from chia.wallet.puzzles.load_clvm import load_clvm
+from chia.wallet.cc_wallet.ccparent import CCParent
+from chia.util.ints import uint64
+from chia.util.hash import std_hash
+
+SINGLETON_MOD = load_clvm("singleton_top_layer.clvm")
+SINGLETON_LAUNCHER = load_clvm("singleton_launcher.clvm")
+SINGLETON_LAUNCHER_HASH = SINGLETON_LAUNCHER.get_tree_hash()
+ESCAPE_VALUE = -113
+MELT_CONDITION = [ConditionOpcode.CREATE_COIN, 0, ESCAPE_VALUE]
+
+
+# Given the parent and amount of the launcher coin, return the launcher coin
+def generate_launcher_coin(coin: Coin, amount: uint64) -> Coin:
+    return Coin(coin.name(), SINGLETON_LAUNCHER_HASH, amount)
+
+
+# Wrap inner puzzles that are not singleton specific to strip away "truths"
+def adapt_inner_to_singleton(inner_puzzle: Program) -> Program:
+    # (a (q . inner_puzzle) (r 1))
+    return Program.to([2, (1, inner_puzzle), [6, 1]])
+
+
+# Take standard coin and amount -> launch conditions & launcher coin solution
+def launch_conditions_and_coinsol(
+    coin: Coin,
+    inner_puzzle: Program,
+    comment: List[Tuple[str, str]],
+    amount: uint64,
+) -> Tuple[Program, CoinSolution]:
```

Comment:
Let's do ` -> Tuple[List[Program], CoinSolution]` and then go `mypy chia tests` to do type checking.